bazel/envoy_binary.bzl

```diff
@@ -59,6 +59,9 @@ def _envoy_linkopts():
         "@envoy//bazel:apple": [
             # https://github.com/envoyproxy/envoy/issues/24782
             "-Wl,-framework,CoreFoundation",
+	    # Required for iOS proxy support.
```

Comment:
This will be included in macOS as well. I think you need to link this in on condition `"@platforms//os:ios"` or something like that.

Reply:
do we actually support macOS as a platform for Envoy?  If someone wanted to use "Envoy Mobile" as a client networking library on macOS, that should be fine too, and we'd want the proxy settings to be applied to them as well.

In general, everything we're selecting on `bazel:apple` throughout the codebase is intended for Envoy Mobile AFAIK.

I'll change the comment to say "Apple proxy support" to keep it general.

Reply:
> do we actually support macOS as a platform for Envoy? 

Yes, it is a supported platform, and has been since before envoy mobile existed

> If someone wanted to use "Envoy Mobile" as a client networking library on macOS, that should be fine too, and we'd want the proxy settings to be applied to them as well.

I'm not sure if that's supported or not; my guess is that we'd need some build system work to get everything setup right.

> In general, everything we're selecting on `bazel:apple` throughout the codebase is intended for Envoy Mobile AFAIK.

That has not always been the case. Please don't assume that!!

Reply:
Thanks for alerting me to this!  I think we should take some inventory of our mobile build rules as a result, but for now, I think this PR is not needed then.  I think the correct place for me to put this is actually in: https://github.com/envoyproxy/envoy/blob/87fa480e31696ac1685d0d4f2eeeb78a41bf410d/mobile/BUILD#L43

I'm going to close this PR and open a new one in mobile to address this.  Thanks again for pointing this out!